qualang_tools/config/waveform_tools.py

```
@@ -88,42 +66,22 @@ def drag_cosine_pulse_waveforms(amplitude, length, alpha, anharmonicity, detunin
-        I_wf = z.real.tolist()  # The `I` component is the real part of the waveform
-        Q_wf = z.imag.tolist()  # The `Q` component is the imaginary part of the waveform
-    else:
-        if alpha != 0 and anharmonicity == 0:
-            raise Exception("Cannot create a DRAG pulse with `anharmonicity=0`")
-        t = np.arange(length, step=1e9 / sampling_rate)  # An array of size pulse length in ns
-        end_point = length - 1e9 / sampling_rate
-        cos_wave = 0.5 * amplitude * (1 - np.cos(t * 2 * np.pi / end_point))  # The cosine function
-        sin_wave = (
-            0.5 * amplitude * (2 * np.pi / end_point * 1e9) * np.sin(t * 2 * np.pi / end_point)
-        )  # The derivative of cosine function
-        z = cos_wave + 1j * 0
-        if alpha != 0:
-            # The complex DRAG envelope:
-            z += 1j * sin_wave * (alpha / (2 * np.pi * anharmonicity - 2 * np.pi * detuning))
-            # The complex detuned DRAG envelope:
-            z *= np.exp(1j * 2 * np.pi * detuning * t * 1e-9)
-        I_wf = z.real.tolist()  # The `I` component is the real part of the waveform
-        Q_wf = z.imag.tolist()  # The `Q` component is the imaginary part of the waveform
+    if alpha != 0 and anharmonicity == 0:
+        raise Exception("Cannot create a DRAG pulse with `anharmonicity=0`")
+    t = np.arange(length, step=1e9 / sampling_rate)  # An array of size pulse length in ns
+    end_point = length - 1e9 / sampling_rate
+    cos_wave = 0.5 * amplitude * (1 - np.cos(t * 2 * np.pi / end_point))  # The cosine function
+    sin_wave = (
+        0.5 * amplitude * (2 * np.pi / end_point * 1e9) * np.sin(t * 2 * np.pi / end_point)
+    )  # The derivative of cosine function
+    z = cos_wave + 1j * 0
+    if anharmonicity != detuning:
+        # The complex DRAG envelope:
+        z += 1j * sin_wave * (alpha / (2 * np.pi * anharmonicity - 2 * np.pi * detuning))
+    # The complex detuned DRAG envelope:
```

same

---

This can be derived if detuning != anharmonicity, but this is the only relevant condition, but I should raise an error otherwise